frontend/ui/lib/widgets/trip_view_widget.dart

```diff
@@ -1,16 +1,42 @@
 import 'package:flutter/material.dart';
+import 'package:tripmeout/model/trip.dart';
+import 'package:tripmeout/services/trip_service.dart';
 import 'package:tripmeout/widgets/place_block_widget.dart';
 
-class TripViewWidget extends StatefulWidget {
-  TripViewWidget({Key key}) : super(key: key);
-  _TripWidgetState createState() => _TripWidgetState();
-}
+class TripViewWidgetFromService extends StatelessWidget {
+  final TripService tripService;
+  final String tripId;
+
+  TripViewWidgetFromService(this.tripService, this.tripId);
 
-class _TripWidgetState extends State<TripViewWidget> {
   @override
-  void initState() {
-    super.initState();
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      future: tripService.getTrip(tripId),
+      builder: (context, snapshot) {
+        if (snapshot.hasData) {
+          return TripViewWidget(snapshot.data);
+        }
+        if (snapshot.hasError) {
+          Scaffold.of(context).showSnackBar(SnackBar(
+            content: Text("Error getting trip"),
+            action: SnackBarAction(
+              label: "Dismiss",
+              onPressed: () {},
```

Comment:
In this case, consider actually adding a "Retry" button that resubmits the fetch-trip request.

Reply:
I will look into actually making this work but Chris said it may be difficulties implementing this. But again I will surely do my research and at least attempt to fix this.